src/net_processing.cpp

```diff
@@ -773,6 +773,26 @@ class PeerManagerImpl final : public PeerManager
     bool AlreadyHaveTx(const GenTxid& gtxid)
         EXCLUSIVE_LOCKS_REQUIRED(cs_main, !m_recent_confirmed_transactions_mutex);
 
+    /**
+     * Map of seen chain tip sets keyed by unique network ID.
+     *
+     * The size of each chain tip set is limited to the number of chain tips in
+     * our global block index.
+     */
+    std::map<uint64_t, std::set<const CBlockIndex*>> m_chain_tip_sets GUARDED_BY(cs_main);
+
+    /** Get the chain tip set for pfrom's network. */
+    std::set<const CBlockIndex*>& GetChainTipSetForPeerNetwork(const CNode& node)
+        EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+
+    /**
+     * Update the chain tip set for a given peer and new block hash by
+     * replacing an existing tip if it is an ancestor of the new block or by
+     * adding a new tip if none of the existing chain tips were replaced.
+     */
+    void UpdateChainTipSet(const CNode& node, const CBlockIndex& potential_new_tip)
```

Comment:
did i understand this correctly? - `UpdateChainTipSet` updates chain tip set based on: 
    1. the whole block/valid headers we received and were told of from other nodes on the same network
    2. when we switch networks, chain tip set initially would only contain [tip of active chain.](https://github.com/bitcoin/bitcoin/blob/86f19bdab78c09a124b15e2d4a72439f13281999/src/net_processing.cpp#L2653)
    3. inbound and outbound connections on same network would have different chain tip set. is this intentional?

if we mine a block and tell other nodes on the network about it, should we update the chain tip set? (i haven't checked if `UpdateChainTipSet` is called in all the locations.) are there other situations in which a block isn’t inserted into the chain tip set?